### ISSUE TITLE
~/.cabal/bin is missing in PATH

### ISSUE THREAD

**Issue by autotaker** [haskell/actions#130]
Hi, 

# Description
I'm using 'haskell/actions/setup@v2' to configure cabal and GHC.
In my workflow, `stylish-haskell` is installed by `cabal install stylish-haskell`, but running the executable fails due to `command not found` error.

I examined the reason, and find that `~/.cabal/bin` is missing in PATH.

Is it possible to add `~/.cabal/bin` to PATH?

# Analysis results
## Before `haskell/actions/setup@v2`

```bash
$ echo $PATH
/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

## After `haskell/actions/setup@v2`

```bash
$ echo $PATH
/home/runner/.ghcup/bin:/home/runner/.ghcup/ghc/9.2.4/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

I think `/home/runner/.cabal/bin` should be added to PATH after the setup step.

## Attachments

- [workflow](https://github.com/autotaker/setup-haskell-path-test/blob/e5676e3ff2599238fd9d6b0e9b874986690056f4/.github/workflows/haskell.yml)
- [Workflow log archive](https://github.com/haskell/actions/files/10105560/setup_log.zip)

# Workarounds
Current workaround for this issue is to source `~/.ghcup/env` in the beginning of each step you want to use package executables.

```yaml
- run: |
    source ~/.ghcup/env
    stylish-haskell
```

However this is annoying because I have to write `source` for every steps.

**Comment by andreasabel** [comment]
I agree, because the default `installdir` of `cabal` is `~/.cabal/bin`:
```
$ cabal --config-file=foo.txt user-config init | grep installdir foo.txt
installdir: $HOME/.cabal/bin
```

I'd like to these alternative workarounds:
1. `cabal user-config update --augment='installdir: ~/.local/bin'`: change the `installdir` to one that is in the path
2. `echo '~/.local/bin' >> ${GITHUB_PATH}`: Add to the `PATH` for the coming workflow `steps`.

**Comment by autotaker** [comment]
@andreasabel 
May I send a PR to implement your workaround?

> `echo '~/.local/bin' >> ${GITHUB_PATH}`: Add to the `PATH` for the coming workflow steps.

**Comment by jneira** [comment]
those workarounds would work in windows?

**Comment by andreasabel** [comment]
> @andreasabel May I send a PR to implement your workaround?
> 
> > `echo '~/.local/bin' >> ${GITHUB_PATH}`: Add to the `PATH` for the coming workflow steps.

I think I mistyped, I meant:
`echo '~/.cabal/bin' >> ${GITHUB_PATH}`

> those workarounds would work in windows?

I'd guess they need refinement to be OS-agnostic.

**Comment by andreasabel** [comment]
@jneira: [It turns out](https://github.com/haskell/actions/actions/runs/3886075764/jobs/6630746511) that for the choco install on Windows, the default installdir is already in the PATH.  Thus, we only need to fix it for Linux and macOS, and then `~/.cabal/bin` should be correct.

**Comment by andreasabel** [comment]
Releasing fix #165 in v2.3.1.
